### PR TITLE
Add support for multiple state machines with different state names

### PR DESCRIPTION
**Pull request by kratob**
I've needed to have more than one state machine within a single model, and ended up adding support to the gem.

Unfortunately, this required a relatively big refactoring, and the code certainly did not get simpler by this.

### CODE REVIEW

**Comment by kratob** [comment]
It's probably wise to wait a bit before merging, until I've validated this actually works well. I'll add it as a `:git` source to my project for the time being.

**Comment by kratob** [comment]
@foobear This is running without issues in production for some weeks now. If you don't veto this, I'll release it as 2.0.0 next week (since it has the breaking change of requiring to include `RailsStateMachine::Model`).

File: README.md [hunk 1]
```diff
@@ -137,6 +104,47 @@ event :request_review do
...
+To use a state attribute other than the default `state`, pass it to the `.state_machine` method:
+
+```
+state_machine :review_state do
+  # ...
+end
+```
+
+This also allows you to define multiple state machines on the same model. Note that event
+names still ahve to be unique for the whole model.
```
Comment (foobear) [inline README.md]: "ahve" => "have"

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,15 +3,17 @@ All notable changes to this project will be documented in this file.
 
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## Unreleased (might warrant major update)
```
Comment (foobear) [inline CHANGELOG.md]: Agree on major version update

**Pull request merged**